app/src/main/java/com/epfl/beatlink/viewmodel/library/PlaylistViewModel.kt

```diff
@@ -184,6 +185,19 @@ class PlaylistViewModel(
     } ?: run { Log.e("PlaylistViewModel", "No playlist selected to update track likes") }
   }
 
+  /**
+   * Create the final list of track to export the playlist that contain the 50 most liked songs in
+   * descending order
+   */
+  fun getFinalListTracks(): List<SpotifyTrack> {
+    return selectedPlaylist_.value
+        ?.playlistTracks
+        ?.sortedByDescending { it.likes } // Sort by likes in descending order
+        ?.take(50) // Take at most 50 tracks
```

Comment:
I would suggest to add the number of tracks as an optional parameter: fun `getFinalListTracks(maxTracks: Int = 50): List<SpotifyTrack>` or to define it as a constant: `private const val DEFAULT_TRACK_LIMIT = 50` (in `Playlist.kt`)